samples/nrf9160/mqtt_simple/src/main.c

```diff
@@ -34,6 +38,31 @@ static bool connected;
 /* File descriptor */
 static struct pollfd fds;
 
+#if !defined(CONFIG_BSD_LIBRARY) && defined(CONFIG_MQTT_LIB_TLS)
+static int certificates_provision(void)
+{
+	static bool certs_added;
+	int err;
+
+	if (!IS_ENABLED(CONFIG_NET_SOCKETS_SOCKOPT_TLS) || certs_added) {
+		return 0;
+	}
+
+	err = tls_credential_add(CONFIG_MQTT_TLS_SEC_TAG,
+				 TLS_CREDENTIAL_CA_CERTIFICATE,
+				 CA_CERTIFICATE,
+				 sizeof(CA_CERTIFICATE));
+	if (err < 0) {
+		printk("Failed to register CA certificate: %d\n", err);
+		return err;
+	}
+
+	certs_added = true;
```

Comment:
Looks like this function is only called once, so the flag is sort of redundant in this use-case

Reply:
Removed the flag